homeassistant/components/keenetic_ndms2/config_flow.py

```diff
@@ -0,0 +1,145 @@
+"""Config flow for Keenetic NDMS2."""
+from ndms2_client import Client, ConnectionException, TelnetConnection
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_SCAN_INTERVAL,
+    CONF_USERNAME,
+)
+from homeassistant.core import callback
+
+from .const import (
+    CONF_CONSIDER_HOME,
+    CONF_INCLUDE_ARP,
+    CONF_INCLUDE_ASSOCIATED,
+    CONF_INTERFACES,
+    CONF_TRY_HOTSPOT,
+    DEFAULT_CONSIDER_HOME,
+    DEFAULT_INTERFACE,
+    DEFAULT_SCAN_INTERVAL,
+    DEFAULT_TELNET_PORT,
+    DOMAIN,
+)
+
+
+class KeeneticFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the options flow for this handler."""
+        return KeeneticOptionsFlowHandler(config_entry)
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        errors = {}
+        if user_input is not None:
+            for entry in self.hass.config_entries.async_entries(DOMAIN):
+                if entry.data[CONF_HOST] == user_input[CONF_HOST]:
+                    return self.async_abort(reason="already_configured")
+
+            _client = Client(
+                TelnetConnection(
+                    user_input[CONF_HOST],
+                    user_input[CONF_PORT],
+                    user_input[CONF_USERNAME],
+                    user_input[CONF_PASSWORD],
+                )
+            )
+
+            try:
+                router_info = await self.hass.async_add_executor_job(
+                    _client.get_router_info
+                )
+
+                return self.async_create_entry(title=router_info.name, data=user_input)
```

Comment:
Please only wrap the line that can raise inside the try... except block. Use an `else:` block if needed.

Reply:
Python magic... cool :+1: